### ISSUE TITLE
No matches returns a 400 - Bad request error

### ISSUE THREAD

**Issue by jimcue** [xiaohutai/jsonapi#70]
A well-formed request that returns no results will get a 400 HTTP code (Bad request) error response.

With no item with id>100 the request `http://example.com/json/articles?filter[id]=>100` gets:

```
{
    "errors": {
        "details": "Bad request: There were no results based upon your criteria!",
        "status": 400,
        "title": "Bad Request"
    }
}
```

Would it not make sense to get a 200 HTTP response with an empty list (or null for specific-item requests) as mentioned in the JSON API specs?
```
HTTP/1.1 200 OK
Content-Type: application/vnd.api+json
{
  "links": {
    "self": "http://example.com/articles"
  },
  "data": []
}
```

**Comment by luminarious** [comment]
204 No Content would be perfect.

**Comment by rwd** [comment]
As mentioned by @jimcue the JSON API spec states that the response code in this scenario **MUST** be `200 OK`: http://jsonapi.org/format/#fetching-resources-responses

**Comment by xiaohutai** [comment]
Fixed via #80.
Will be available in a new release later today (or tomorrow).

**Comment by xiaohutai** [comment]
Should be fixed in [v3.2.0](https://github.com/xiaohutai/jsonapi/releases/tag/v3.2.0)
